Formula/quantum++.rb

```diff
@@ -0,0 +1,44 @@
+class Quantumxx < Formula
+  desc "Modern C++ quantum computing library"
+  homepage "https://github.com/softwareQinc/qpp"
+  url "https://github.com/softwareQinc/qpp/archive/v4.1.tar.gz"
+  sha256 "b2d0f84a4726be71e45dc03ad0556ff22f7e6b92c19676391899e1df9091c061"
+  license "MIT"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "eigen"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    (testpath/"CMakeLists.txt").write <<~EOS
+      cmake_minimum_required(VERSION 3.15)
+      project(qpp_test)
+      set(CMAKE_CXX_STANDARD 17)
+
+      find_package(qpp REQUIRED)
+      add_executable(qpp_test qpp_test.cpp)
+      target_link_libraries(qpp_test PUBLIC ${QPP_LINK_DEPS} libqpp)
+    EOS
+    (testpath/"qpp_test.cpp").write <<~EOS
+      #include <iostream>
+      #include <qpp/qpp.h>
+
+      int main() {
+          using namespace qpp;
+          std::cout << disp(transpose(0_ket)) << std::endl;
+      }
+    EOS
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
```

Comment:
I think it's worth considering whether we should just compile `qpp_test.cpp` without relying on CMake (so we don't need is as a `:test` dependency).

Reply:
That's not really possible. `cmake` will just call the compiler to build things, and we can always run the exact same command that `cmake` will.